Quiddity, RequestId & Uuid deprecation

jamessimone opened the pull request:
* Fixes #27 by taking advantage of Winter 21 features. 
* Removes Uuid dependency
* Small QOL logging improvements related to Quiddity granularity

jamessimone commented:
@jongpie this can only be tested on sandboxes with Winter 21 till it's been fully released!

jamessimone commented on nebula-logger/main/callouts/classes/RestApi.cls:
```diff
@@ -5,7 +5,7 @@
 
     private static Boolean ALL_OR_NONE_DEFAULT       = true;
-    private static String BASE_URL                   = System.Url.getSalesforceBaseUrl().toExternalForm() + '/services/data/v49.0';
+    private static String BASE_URL                   = System.Url.getSalesforceBaseUrl().toExternalForm() + '/services/data/v50.0';
```
I was considering making this programmatic a la the test - your thoughts?

jongpie commented:
@jamessimone I added a couple of comments, but overall this looks great! Let's wait until next weekend to merge, once all environments have been upgraded to Winter '21.

jamessimone commented on sfdx-project.json:
```diff
@@ -2,10 +2,13 @@
     "packageDirectories": [
         {
             "path": "nebula-logger",
+            "package": "NebulaLogger",
```
Makes sense, sure!

jamessimone commented on nebula-logger/main/logger/classes/Logger.cls:
```diff
@@ -4,10 +4,9 @@
 
-    public enum SaveMethod { DML, REST_API, QUEUEABLE_JOB }
-
+    private static final Request REQUEST = System.Request.getCurrent();
```
Yeah, I can remove the storage of it, I agree reducing heap size makes the slight repetition worth it

jamessimone commented:
> @jamessimone I added a couple of comments, but overall this looks great! Let's wait until next weekend to merge, once all environments have been upgraded to Winter '21.

@jongpie sounds good. I rebased and added two commits, 5914f1cb51321703111bc0326b4f1806cacf4237 I think you'll like in particular for taking advantage of the safe navigation feature also new in Winter '21. It's possible that there are other instances in the codebase that can also take advantage of safe navigation to reduce boilerplate, as I haven't performed a complete scan.

I'll merge this PR next Sunday!

jongpie commented on nebula-logger/main/logger/classes/LogEntryBuilder.cls:
```diff
@@ -171,25 +171,25 @@ public without sharing virtual class LogEntryBuilder {
-        this.logEntryEvent.LoginHistoryId__c                 = sessionMap == null ? null : sessionMap.get('LoginHistoryId');
-        this.logEntryEvent.LoginType__c                      = sessionMap == null ? null : sessionMap.get('LoginType');
-        this.logEntryEvent.LogoutUrl__c                      = sessionMap == null ? null : sessionMap.get('LogoutUrl');
+        this.logEntryEvent.LoginDomain__c                    = sessionMap?.get('LoginDomain');
```
I just realized you're using the new safe navigation now for null checks - love it!

jamessimone replied:
it's a huge win for readability!
